compliance_checker/cf/cf_1_7.py

```diff
@@ -724,17 +728,105 @@ def _evaluate_towgs84(self, val):
+            test_ctx.messages.append("Not all of the grid_mapping variables exist in the file")
+            test_ctx.out_of += 1
+        
+        test_ctx.score += 1
+         
+        # Get the CF coordinate variables or auxiliary coordinate variables 
+        auxiliary_coordinate_variables = cfutil.get_auxiliary_coordinate_variables(ds)
+        coordinate_variables = cfutil.get_coordinate_variables(ds) 
+        # Check the grid_mapping variable coordinates if listed.
+            # [4/9] Each coord_var is the name of a coordinate variable or 
+            # auxiliary coordinate variable, which must exist in the file. If it 
+            # is an auxiliary coordinate variable, it must be listed in the 
+            # coordinates attribute.    
+        for coord_item in grid_mapping_variables_coordinates:                                                   
+            if coord_item not in coordinate_variables and coord_item not in auxiliary_coordinate_variables:
+                test_ctx.messages.append("{} is not listed as coordinate or auxiliary coord Variable.".format(coord_item))
+                test_ctx.out_of += 1
+        
+        test_ctx.score += 1       
+
+        # Check the gid_mapping variable attributes
+        for var_name in sorted(grid_mapping_variables):           
             var = ds.variables[var_name]
             test_ctx = self.get_test_ctx(
-                BaseCheck.HIGH, self.section_titles["5.6"], var.name
-            )
+                BaseCheck.HIGH, self.section_titles["5.6"], var.name)
+            
+            # Recommendation: The grid mapping variables should have 0 dimensions.
+            if var.dimensions:
+               test_ctx.messages.append("The grid mapping variable {} should have 0 dimension".format(var.name))
+               test_ctx.out_of += 1
+            
+            test_ctx.score += 1
+
+            # [5/9] The grid mapping variables must have the grid_mapping_name attribute. 
+            # The legal values for the grid_mapping_name attribute are contained in Appendix F.
+            if "grid_mapping_name" not in var.ncattrs():
+                test_ctx.messages.append("The grid mapping variable "
+                "{} doesn't have the grid_mapping_name attribute".format(var.name))
+                test_ctx.out_of += 1
+            else:
+                if var.grid_mapping_name not in grid_mapping_dict17:
+                   test_ctx.messages.append("The legal values for the grid_mapping_name attribute "
+                    "{} is not contained in Appendix F".format(var.grid_mapping_name))             
+                   test_ctx.out_of += 1
+                
+            test_ctx.score += 1
+
+            # [6/9] The data types of the attributes of the 
+            # grid mapping variable must be specified in Table 1 of Appendix F.
+            type_map = {"S": "str" , "N": "float64"}
```

Comment:
The typecheck for `N` is likely too strong.  `N` simply means that the value has a numeric type, not just floating point.

Reply:
agreed.... changed to:
type_map = {"S": "str", "N": ["float64", "int"]}